searx/engines/gitea.py

```diff
@@ -0,0 +1,57 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+"""Github (IT)
+
+"""
+
+from urllib.parse import urlencode
+from dateutil import parser
+
+about = {
+    "website": 'https://about.gitea.com',
+    "wikidata_id": None,
+    "official_api_documentation": 'https://docs.gitea.com/next/development/api-usage',
+    "use_official_api": True,
+    "require_api_key": False,
+    "results": 'JSON',
+}
+
+categories = ['it', 'repos']
+paging = True
+
+base_url = "https://codeberg.org"
```

Comment:
Lets set default `base_url` to `None`  and add a [`init(..)`](https://docs.searxng.org/dev/engines/demo/demo_online.html#searx.engines.demo_online.init)  to check if base_url is configured in the engine settings.